src/disk.c

```diff
@@ -63,6 +63,140 @@ static PedExceptionOption disk_exception_handler(PedException* ex) {
 	return PED_EXCEPTION_UNHANDLED;
 }
 
+static char *blk_device_by_path(const gchar* path) {
+	struct stat st = { 0 };
+
+	if (!path) {
+		LOG(MOD "Path is empty\n");
+		return NULL;
+	}
+	if (stat(path, &st) != 0) {
+		LOG(MOD "Cannot stat '%s'\n", path);
+		return NULL;
+	}
+
+	return blkid_devno_to_devname(st.st_dev);
+}
+
+static PedPartition *get_rootfs_partition(PedDisk* disk) {
+	char *rootfs_dev_path = NULL;
+	char *part_path = NULL;
+	PedPartition *part = NULL;
+	int last_partition_num = 0;
+	int i = 0;
+
+	rootfs_dev_path = blk_device_by_path("/");
```

Comment:
This returns an allocated pointer, needs to be freed after use?

Reply:
yes, here https://github.com/clearlinux/micro-config-drive/pull/35/files#diff-352f7d19b14cf77c9be0554e611634faR117